Realm/RLMLogger.h

```diff
@@ -48,33 +48,95 @@ typedef RLM_CLOSED_ENUM(NSUInteger, RLMLogLevel) {
     RLMLogLevelAll
 } NS_SWIFT_NAME(LogLevel);
 
+/**
+ An enum representing different categories of sync-related logging that can be configured.
+ Setting the log level for a parent category automatically sets the same level for all child categories.
+ Category hierarchy:
+```
+ Realm
+ ├─► Storage
+ │   ├─► Transaction
+ │   ├─► Query
+ │   ├─► Object
+ │   └─► Notification
+ ├─► Sync
+ │   ├─► Client
+ │   │   ├─► Session
+ │   │   ├─► Changeset
+ │   │   ├─► Network
+ │   │   └─► Reset
+ │   └─► Server
+ ├─► App
+ └─► Sdk
+```
+*/
+typedef NS_ENUM(NSUInteger, RLMLogCategory) {
+    ///  Top level log category for Realm, updating this category level would set all other subcategories too.
+    RLMLogCategoryRealm,
```

Comment:
[nit] This naming scheme is causing a bit of tautology as we have the `RLM` prefix and the `Realm` suffix/infix. One option would be to have

```
RLMLogCategoryAll,
RLMLogCategorySDK,
...
```

Essentially dropping the `Realm` as it's part of every category, thus not bringing any new information. Not a blocker though.

Reply:
Should we change RLMLogCategoryRealm as well?, given that this is the keyword we use for all categories in all SDKs?

Reply:
Yeah, I feel a bit ambiguous about it - I'd be fine with either keeping it as `RLMLogCategoryRealm` or with replacing it with `RLMLogCategoryAll`. I think in other SDKs we can get away with it because we don't have the `RLM` prefix (similarly to what we do in Swift). That being said, perhaps we shouldn't overthink it as Obj C users are so few and whether we have `Realm` in there or not is unlikely to make a huge difference.

Reply:
In Swift we have Category.realm so to keep uniformity in both APIs, we can keep it like that, and then remove Realm for the rest of the categories

Reply:
Added changes